### ISSUE TITLE
[FEATURE]: Add solution to Q31 to Q40 of Leetcode

### ISSUE THREAD

**Issue by DugarRishab** [DugarRishab/Algo.io#23]
# Description: 

- Add the solution of Q31 to Q40 of Leetcode in any language. 

# Questions: 

- [x] Q30 - cpp
- [x] Q31 - cpp
- [ ] Q32
- [x] Q33 - java
- [x] Q34 - cpp
- [x] Q35
- [x] Q36
- [ ] Q37
- [ ] Q38
- [x] Q39
- [x] Q40 - cpp

# Eligibility Criteria: 

- The solution must not already be present using the same language. If you want to add an alternative to any solution that is already present then mark it as `Alternative`
- You can provide the solution to any question in a different language than what is already used. 
- The solution must pass all tests on Leetcode and do provide a screenshot proving the same. 
- NOTE: Get yourself assigned before sending a PR. PRs regarding this issue from non-assigned people will be marked `spam`

**Comment by raginiisingh** [comment]
Hey, can you please assign me this issue

**Comment by DugarRishab** [comment]
Hey @raginiisingh. I will assign this to you now.

**Comment by vivek-dhiman** [comment]
@DugarRishab I would like to add solution to Q33. Please assign the issue to me.

**Comment by DugarRishab** [comment]
> @DugarRishab I would like to add solution to Q33. Please assign the issue to me.

Assigning you now

**Comment by aryanayush012** [comment]
Please Assign this to me @DugarRishab

**Comment by ShubhamKumar47** [comment]
Please assign this to me @DugarRishab

**Comment by DugarRishab** [comment]
@ShubhamKumar47 and @aryanayush012 assigning you now

**Comment by omsurase** [comment]
hey @DugarRishab, could you please assign me question 40

**Comment by DugarRishab** [comment]
> hey @DugarRishab, could you please assign me question 40

Yeah sure

**Comment by sarthakw7** [comment]
Hey @DugarRishab ,  I would like to add solution to Q35. Please assign the issue to me.

**Comment by anirudh0806** [comment]
Hey @DugarRishab , I would like to add solution to Q39. Please assign the issue to me.

**Comment by omsurase** [comment]
@DugarRishab is there a question list or am I free to pick a question that I find interesting?

**Comment by rahul-jamwal** [comment]
Hey @DugarRishab I would like to do Q36. Could you please assign this to me

**Comment by DugarRishab** [comment]
> @DugarRishab is there a question list or am I free to pick a question that I find interesting?

Yes you are free to choose a question. But the solution must not already be there.

**Comment by DugarRishab** [comment]
> Hey @DugarRishab I would like to do Q36. Could you please assign this to me

Yes,  I will assign you now.

**Comment by DugarRishab** [comment]
@sarthakw7 @anirudh0806 sorry for not responding earlier. I will assign you now. But as you can see, there are 6 questions left and 8 contenders. So solutions will be accepted on fcfs basis. 
@vivek-dhiman @aryanayush012 @raginiisingh @rahul-jamwal @omsurase and @ShubhamKumar47

**Comment by RP2025** [comment]
I would like to contribute to Q35 with python, if available kindly assign me this

**Comment by DugarRishab** [comment]
> I would like to contribute to Q35 with python, if available kindly assign me this

Yeah, go ahead

**Comment by RP2025** [comment]
I would like to add the solution of 38 using python.
kindly assign me this

**Comment by dhullrajat** [comment]
Hii, @DugarRishab ,
Can you please assign me questions 32 and 37? 
I am good at explaining codes via comments.

**Comment by DugarRishab** [comment]
Hey, @RP2025 and @dhullrajat I will assign you both now

**Comment by MehakGupta1103** [comment]
Can you please assign question number 38 to me, i eagerly want to contribute to this one

**Comment by DugarRishab** [comment]
Yeah, @MehakGupta1103 go ahead and contribute

**Comment by Sakshi2002-Sinha** [comment]
Can you assign me this issue?
If any of the questions are left.

**Comment by DugarRishab** [comment]
Hi @Sakshi2002-Sinha, i will assign this issue to you now.

**Comment by Sakshi2002-Sinha** [comment]
> Hi @Sakshi2002-Sinha, i will assign this issue to you now.

I have made a PR. Please check and tell me if there are further any changes needed.

**Comment by bhawarth-garg** [comment]
can you assign me this issue? @DugarRishab 
I would like to add detailed solution for Q31,33,39.

**Comment by garg-Megha** [comment]
@DugarRishab  I would like to add a solution in c++ for problems 32 and 37. Please assign this task to me.

**Comment by MehakGupta1103** [comment]
[reference](https://github.com/DugarRishab/Algo.io/issues/23#issuecomment-1278974278 )
I have added the solution to the problem 38, @DugarRishab please review it #86 and kindly merge my pr , thankyou

**Comment by kshavgupta** [comment]
Hey! @DugarRishab , please assign this issue to me.
